src/spec-node/featuresCLI/publishCommandImpl.ts

```diff
@@ -0,0 +1,92 @@
+import { fetchRegistryAuthToken, HEADERS } from '../../spec-configuration/containerFeaturesOCI';
+import { request } from '../../spec-utils/httpRequest';
+import { Log, LogLevel } from '../../spec-utils/log';
+
+const semverCompare = require('semver-compare');
+const semver = require('semver');
+
+interface versions {
+    name: string;
+    tags: string[];
+}
+
+export async function getPublishedVersions(featureId: string, registry: string, namespace: string, output: Log) {
+    try {
+        const url = `https://${registry}/v2/${namespace}/${featureId}/tags/list`;
+        const resource = `${registry}/${namespace}/${featureId}`;
```

Comment:
I would create a `OCIFeatureRef` object early on if possible so that we can pass that around instead of strings.

Reply:
Done